test/test_nn.py

```diff
@@ -2192,6 +2192,21 @@ def test_RNN_cell(self):
 
                 hx.sum().backward()
 
+    def test_RNN_cell_no_broadcasting(self):
+        def test(cell_module, hx, output_size):
+            input = Variable(torch.randn(3, 10))
+            cell = cell_module(10, hidden_size)
+            self.assertRaises(RuntimeError, lambda: cell(input, hx))
+
+        hidden_size = 20
+        bad_hx = Variable(torch.randn(1, hidden_size))
+        good_hx = Variable(torch.randn(3, hidden_size))
+
+        test(nn.RNNCell, bad_hx, hidden_size)
+        test(nn.GRUCell, bad_hx, hidden_size)
+        test(nn.LSTMCell, (bad_hx, good_hx), hidden_size)
+        test(nn.LSTMCell, (good_hx, bad_hx), hidden_size)
```

Comment:
What if `hidden_size == 1`? Does this fail as well?

Reply:
Yes, `hidden_size == 1` with bad inputs throws an error. Added this case to the test